### PR TITLE
Add analytical event to notification

### PR DESCRIPTION
**Pull request by RobinDevelopersNL**

### CODE REVIEW

File: src/js/components/molecules/in-page-notification/in-page-notification.ts [hunk 1]
```diff
@@ -2,16 +2,41 @@
...
     public display: boolean = true;
+    private $emitPublic;
+    private $route;
 
     public close(): void {
         this.display = false;
+        this.triggerEventClick();
+    }
+
+    public clickEventData(): {} | null {
```
Comment (dan-entous) [inline src/js/components/molecules/in-page-notification/in-page-notification.ts]: personally, i prefere a guard clause:
```
if (!this.clickEvent) {
    return null;
}
```

File: src/js/components/molecules/in-page-notification/in-page-notification.ts [hunk 2]
```diff
@@ -2,16 +2,41 @@
...
+                data: {
+                    ...data,
+                    pageSlug: this.$route.fullPath,
+                },
+            };
+        }
+        return null;
+    }
+
+    public triggerEventClick(): void {
```
Comment (dan-entous) [inline src/js/components/molecules/in-page-notification/in-page-notification.ts]: same comment re: guard clause:
```
if (!this.clickEvent) {
    return;
}
````

File: src/js/components/molecules/in-page-notification/in-page-notification.ts [hunk 3]
```diff
@@ -2,16 +2,41 @@
...
 
 @WithRender
 @Component({
     name: 'InPageNotification',
 })
 export class InPageNotification extends Vue implements NotificationInterface {
-    @Prop({ type: String, required: true }) message!: NotificationInterface['message'];
+    @Prop({ type: String, required: true }) public message!: NotificationInterface['message'];
+    @Prop({ type: Object, required: false }) public clickEvent?: NotificationInterface['clickEvent'];
     public display: boolean = true;
```
Comment (dan-entous) [inline src/js/components/molecules/in-page-notification/in-page-notification.ts]: 🐜 new line between @Prop and class variable

File: src/js/components/molecules/in-page-notification/in-page-notification.ts (hunk 1 above)
Comment (RobinDevelopersNL) [reply src/js/components/molecules/in-page-notification/in-page-notification.ts]: You are right Dan, I will modify it.

File: src/js/components/molecules/in-page-notification/in-page-notification.ts (hunk 2 above)
Comment (RobinDevelopersNL) [reply src/js/components/molecules/in-page-notification/in-page-notification.ts]: You are right Dan, I will modify it.

File: src/js/components/molecules/in-page-notification/in-page-notification.ts (hunk 3 above)
Comment (jasperkuipers) [reply src/js/components/molecules/in-page-notification/in-page-notification.ts]: i agree

File: src/js/components/molecules/in-page-notification/in-page-notification.ts (hunk 1 above)
Comment (jasperkuipers) [reply src/js/components/molecules/in-page-notification/in-page-notification.ts]: 👍

**Pull request merged**